### DISCUSSION TITLE
An `@auth/astro` package? Support for the AstroJS framework.

### DISCUSSION THREAD

**Discussion by JulianCataldo** [nextauthjs/next-auth#6288] [Ideas]
Hello there.

Based on the SvelteKit core wrapper:
https://github.com/nextauthjs/next-auth/tree/main/packages/frameworks-sveltekit

It should be possible to make an Astro SSR mode wrapper for Auth.js.
However I'm not comfortable enough with this project to judge the complexity of this task.

There is an existing `@astro-auth` project, from which I've shared the idea of joining effort.
https://github.com/astro-community/astro-auth/issues/27

So I would like to know if some others are interested by this.
Note : for now `@astro-auth` works well for what I do, so I don't have much incentive to dig the above topic, yet.

Cheers.

**Comment by tajkirkpatrick** [comment]
@astro-auth failed me today, which motivated me to begin this fork repo and try to get to a proof of concept. For any others that may stumble upon this discussion, I would appreciate your assistance as I've never contributed to OSS before.

If confused, under the astro-auth branch see "/packages/frameworks-astro/Readme.md", 
Disclaimer: that is a full astrojs project in previously mentioned folder structure, NOT just javascript functions to be exported and used immediately.

https://github.com/tajkirkpatrick/next-auth/tree/astro-auth

**Reply by tajkirkpatrick** [reply]
attempting to tackle it, I opened #6373 after running into an error with the Crypto WebAPI. Here's a replication.
https://stackblitz.com/edit/node-mkewjo?file=src/libs/index.ts
